ReactWindows/ReactNative.Net46/Views/Text/ReactTextShadowNode.cs

```diff
@@ -238,6 +275,7 @@ private void UpdateTextBlockCore(TextBlock textBlock, bool measureOnly)
             textBlock.FontSize = _fontSize ?? 15;
             textBlock.FontStyle = _fontStyle ?? new FontStyle();
             textBlock.FontWeight = _fontWeight ?? FontWeights.Normal;
+            textBlock.TextDecorations = new TextDecorationCollection(TextDecorationCollection);
 
```

Comment:
My guess would be you may need to create virtual node types and control all this by checking for presence of the prop in JavaScript.  I.e., in JavaScript, you might check:
```
if (this.props.textDecorationLine === 'underline') {
  this.props.children = (<UnderlinedText>{this.props.children}</UnderlinedText>);
}
```
or something along these lines. I'm sure my syntax is a bit off.